design/ceph/ceph-telemetry.md

```diff
@@ -0,0 +1,133 @@
+# Adding Rook data to Ceph telemetry
+The solution plan agreed upon with the telemetry team is for the Rook operator to add telemetry to
+the Ceph mon `config-key` database, and Ceph will read each of those items for telemetry retrieval.
+
+## Guidelines from the Ceph telemetry team
+- Users must opt in to telemetry
+- Users must also opt into the new (delta) telemetry items added between Ceph versions
+- Rook should not add information to `config-key` keys that can grow arbitrarily large to keep space
+  usage of the mon database low (limited growth is still acceptable)
+
+## Metrics for Rook to collect
+Metric names will indicate a hierarchy that can be parsed to add it to  Ceph telemetry collection in
+a more ordered fashion.
+
+For example `rook/version` and `rook/kubernetes/version` should be put into a structure like shown:
+```json
+"rook": {
+  "version": "vx.y.z"
+  "kubernetes": {
+    "version": "vX.Y.Z"
+  }
+}
+```
+
+### Overall metrics
+- `rook/version` - Rook version.
+- `rook/kubernetes/...`
+  - `rook/kubernetes/version` - Kubernetes version.
+    - Ceph already collects os/kernel versions.
```

Comment:
I'm curious if Ceph actually collects these in a K8s environment as expected. Or maybe you said you already verified that.

Reply:
You can check out the sample I got from a minikube test cluster: https://gist.github.com/BlaineEXE/2fafaaeec065803390ebd75b498c1a59

I think it largely only gathers the in-container information and not information about kubernetes' bare hosts. The must-gather has CentOS Stream 8 as the OS, which definitely isn't what minikube uses, but is I think what the Ceph images use.

Currently, I don't know if there would be a good way to collect the actual raw host info, but it would seem useful if we can. I would leave this for a follow-up unless someone has good ideas for collecting the info.